prototype2/common/Producer.h

```diff
@@ -10,14 +10,24 @@
 #pragma once
 
 #include <librdkafka/rdkafkacpp.h>
+#include <common/Buffer.h>
+#include <functional>
 
 
 ///
 class ProducerBase {
 public:
   ProducerBase() = default;
   virtual ~ProducerBase() = default;
-  virtual int produce(char *buffer, int length) = 0;
+
+  // \todo deprecate this function in favor of encapsulated buffer
+  virtual int produce(void* buffer, size_t bytes) = 0;
+
+  template<typename T>
+  inline void produce2(const Buffer<T>& buffer)
```

Comment:
why not use the same name produce() ? The arguments are different so should not bother the compiler

Reply:
Could be. I was tempted to just deprecate the old one and refactor everything to use Buffer, but I was not sure how you would like it. If you are ok with it, I can go all the way with it while I'm at it?

Reply:
sure go ahead